Py_FS/wrapper/nature_inspired/AROA.py

```diff
@@ -0,0 +1,199 @@
+"""
+Programmer: Soumitri Chattopadhyay
+Date of Development: 11/07/2021
+This code has been developed according to the procedures mentioned in the following research article:
+"Hashim, F.A., Hussain, K., Houssein, E.H. et al. Archimedes Optimization Algorithm.
+Applied Intelligence, 51, 1531–1551 (2021)"
+"""
+
+import numpy as np
+from sklearn import datasets
+
+from Py_FS.wrapper.nature_inspired.algorithm import Algorithm
+
+
+class AROA(Algorithm):
+
+    # Archimedes Optimization Algorithm
+    ############################### Parameters ####################################
+    #                                                                             #
+    #   num_agents: number of agents                                              #
+    #   max_iter: maximum number of generations                                   #
+    #   train_data: training samples of data                                      #
+    #   train_label: class labels for the training samples                        #
+    #   obj_function: the function to maximize while doing feature selection      #
+    #   trans_function_shape: shape of the transfer function used                 #
+    #   save_conv_graph: boolean value for saving convergence graph               #
+    #                                                                             #
+    ###############################################################################
+
+    def __init__(self,
+                 num_agents,
+                 max_iter,
+                 train_data,
+                 train_label,
+                 save_conv_graph=False,
+                 trans_func_shape='s',
+                 seed=0):
+
+        super().__init__(num_agents=num_agents,
+                         max_iter=max_iter,
+                         train_data=train_data,
+                         train_label=train_label,
+                         save_conv_graph=save_conv_graph,
+                         trans_func_shape=trans_func_shape,
+                         seed=seed)
+
+        self.algo_name = 'AROA'
+        self.agent_name = 'Particle'
+        self.algo_params = {}
+
+    def user_input(self):
+        # initializing parameters
+        self.algo_params['C1'] = 2
+        self.algo_params['C2'] = 6
+        self.algo_params['C3'] = 2
+        self.algo_params['C4'] = 0.5
+        self.algo_params['upper'] = 0.9
+        self.algo_params['lower'] = 0.1
+
```

Comment:
Allow the user to enter the values of the parameters [Refer to the user_input() method of WOA_test.py].

Reply:
Done, ranges/default values given as in relevant paper